File: 9 Palindrome Number/solution.py
class Solution:
    # @param {integer} x
    # @return {boolean}
    def isPalindrome(self, x):
        result = 0
        tmp = x
        while (tmp>0):
            tmp,rem = divmod(tmp,10)
            result = result*10+rem
        return result==x
        

File: 9 Palindrome Number/solution.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        queue<int> helper;
        int tmp = x;
        while (tmp>0){
            int digit = tmp%10;
            helper.push(digit);
            tmp = tmp/10;
        }
        int result = 0;
        int len = helper.size();
        for (int i=0;i<len;i++){
            int revdig = helper.front();
            helper.pop();
            result = result*10+revdig;
        }
        return result==x;
    }
};